pkg/util/kubernetes/apiserver/metadata_controller.go

```diff
@@ -320,7 +316,10 @@ func GetPodMetadataNames(nodeName, ns, podName string) ([]string, error) {
 	return metaList, nil
 }
 
-// GetNamespaceLabels retrieves the labels of the queried namespace from the cache of the shared informer.
+// GetNamespaceLabels retrieves the labels of the queried namespace.
```

Comment:
I've noticed that this function is called from the node agent when the Cluster Agent is not enabled, but I'm not sure if it's supposed to work. It doesn't work for me on the latest version. I decided to leave it here for now, but maybe we can remove it.